app.ts

```diff
@@ -0,0 +1,230 @@
+	}
+	return { firstNumber, lastNumber };
+}
+
+function updateArrowVisibility(firstNumber: number, lastNumber: number, recordCount: number) {
+	if (firstNumber === 0) {
+		$('.arrow-left').hide();
+	} else {
+		$('.arrow-left').show();
+	}
+
+	if (lastNumber >= recordCount) {
+		$('.arrow-right').hide();
+	} else {
+		$('.arrow-right').show();
+	}
+}
+
+function fetchAndDisplayRecords(firstNumber: number, lastNumber: number, inputValue: number) {
+	return myClass.fetchRecords(firstNumber, lastNumber)
+		.then(records => {
+			const tbody = $("tbody");
+			tbody.empty();
+			for (const record of records) {
+				/** creates row for each record*/
+				$("tbody").append(`<tr class="row"></tr>`);
+				const lastRow = $(".row:last");
+				for (const value of record) {
+					/** assign each record to their column in a specified row */
+					lastRow.append(`<td>${value}</td>`);
+				}
+				if (record.includes(inputValue)) {
+					/** highlights the searched row */
+					lastRow.css('background-color', '#DDC0B4');
+				}
+				$("tbody").append(lastRow);
+			}
+			return records;
+		})
+		.catch(err => {
+			throw new Error('Error fetching records' + err);
+		});
+}
+
+/** recalculates the record range that includes inputValue */
+function searchRecordsAndResize(recordCount: number) {
+	let inputValue = $('#searchInput').val() as number;
+	if (inputValue < 0 || inputValue > recordCount) {
+		$('.modal').css('display', 'block');
+		$('.content').append(`<p>${inputValue} is not a number within the range. Please try a different number</p>`);
+		$('#page').empty().append(`Showing record: ${myClass.firstNumber} - ${myClass.lastNumber}`);
+		$('#searchInput').val('');
+		return;
+	}
+	let calculatedRows = calculatingRows();
+	/** divides the calculated max rows in half*/
+	const halfRange = Math.floor(calculatedRows / 2);
+	myClass.firstNumber = Math.max(0, inputValue - halfRange);
+	myClass.lastNumber = Math.min(recordCount, myClass.firstNumber + (calculatedRows - 1));
+	displayRecords(recordCount);
+}
+
+/** Navigates to the next set of records */
+function rightArrow(recordCount: number) {
+	$('#page').empty();
+	$('#searchInput').val('');
+	/** retrieves the last row */
+	const lastRow = document.querySelector("#recordsTable tbody .row:last-child");
+	/** checks if the last row exists */
+	if (lastRow) {
+		const cells = lastRow.querySelectorAll("td");
+		const lastRecord = [];
+		for (const cell of Array.from(cells)) {
+			lastRecord.push(cell.textContent || "");
+		}
+		/** determines te value in the last row */
+		const lastID = parseFloat(lastRecord[0]);
+		/** checks if the last value is within range */
+		if (0 <= lastID && lastID <= recordCount) {
+			$("tbody").empty();
+			/** calculates the first number of the page */
+			myClass.firstNumber = lastID + 1;
+			let calculatedRows = calculatingRows();
+			/** calculates the last number of the page */
+			myClass.lastNumber = myClass.firstNumber + (calculatedRows - 1);
+			displayRecords(recordCount);
```

Comment:
With this function not returning anything, you are going to have to handle the error that `displayRecords` can throw.